.github/workflows/spread-docs.yaml

```diff
@@ -0,0 +1,45 @@
+name: Spread tests for docs
+on:
+  pull_request:
+    paths:
+      - spread.yaml
+      - docs/**/code/**
+  push:
+    branches:
+      - main
+    paths:
+      - spread.yaml
+      - docs/**/code/**
+  schedule:
+    - cron: "0 0 * * 0" # Midnight UTC on Sundays
+
+jobs:
+  docs-tests:
+    runs-on: spread-installed
+    steps:
+      - name: Cleanup job workspace
+        run: |
+          rm -rf "${{ github.workspace }}"
+          mkdir "${{ github.workspace }}"
+
+      - name: Checkout charmcraft
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          submodules: true
+
+      - name: Download snap from store
+        run: |
+          snap download --edge charmcraft
```

Comment:
I'm afraid we'll eventually reach a point where the doc tests need the PR's version of charmcraft, which will make this quite hard to debug
I don't really know how to fix this though, short of packing the snap again here (for the 4th or so time probably)

Reply:
I agree, but I'm kicking that can down the road for now because it's something I'd like to eventually solve with a workflow that starts by building the snap and then conditionally kicks off every job that consumes the snap, ideally using Starflow to do this for all our apps.